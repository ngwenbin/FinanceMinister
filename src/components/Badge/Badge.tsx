import { Text, TextStyle, View, ViewStyle } from "react-native";
import { Colors } from "@constants/Colors";
import { getFont } from "@utils";
import { testIds } from "@utils/tests/testIds";

type BadgeVariant = "green" | "blue" | "red" | "yellow";

const badgeVariants: Record<BadgeVariant, ViewStyle & TextStyle> = {
  green: {
    color: Colors.success[700],
    backgroundColor: Colors.success[100],
  },
  blue: {
    color: Colors.info[700],
    backgroundColor: Colors.info[100],
  },
  red: {
    color: Colors.error[700],
    backgroundColor: Colors.error[100],
  },
  yellow: {
    color: Colors.warning[700],
    backgroundColor: Colors.warning[100],
  },
};

interface BadgeProps {
  label: string;
  variant: BadgeVariant;
  rounded?: boolean;
}

const Badge = ({ label, variant, rounded = true }: BadgeProps) => {
  return (
    <View
      testID={testIds.sharedComponents.badge.badgeContainer}
      style={{
        backgroundColor: badgeVariants[variant].backgroundColor,
        borderRadius: rounded ? 12 : 6,
        paddingHorizontal: 8,
        paddingVertical: 4,
      }}
    >
      <Text
        testID={testIds.sharedComponents.badge.badgeLabel}
        style={{
          color: badgeVariants[variant].color,
          ...getFont("medium", "xs"),
        }}
      >
        {label}
      </Text>
    </View>
  );
};

export default Badge;
