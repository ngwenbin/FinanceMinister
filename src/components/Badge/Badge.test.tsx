import "@testing-library/jest-native/extend-expect";
import { render, screen } from "@testing-library/react-native";
import { testIds } from "@utils/tests/testIds";
import Badge from "./Badge";

const badgeContent = "This is a text";
const badgeLabelTestId = testIds.sharedComponents.badge.badgeLabel;

describe("<Badge>", () => {
  describe("as default", () => {
    test("Renders successfully", () => {
      const activeText = render(<Badge label={badgeContent} variant="blue" />);
      expect(activeText).toBeDefined();
    });

    test("Renders correctly", () => {
      const activeText = render(<Badge label={badgeContent} variant="blue" />);
      expect(activeText.toJSON()).toMatchSnapshot();
    });

    test("Outputs the correct text", () => {
      render(<Badge label={badgeContent} variant="blue" />);
      expect(screen.getByTestId(badgeLabelTestId)).toHaveTextContent(
        badgeContent
      );
    });
  });
});
