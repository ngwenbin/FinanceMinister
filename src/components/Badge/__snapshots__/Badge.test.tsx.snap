// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge> as default Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#DDEFFE",
      "borderRadius": 12,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
    }
  }
  testID="badge-0"
>
  <Text
    style={
      Object {
        "color": "#2C56B3",
        "fontFamily": "Inter_500Medium",
        "fontSize": 12,
      }
    }
    testID="badge-1"
  >
    This is a text
  </Text>
</View>
`;
